pkg/apis/calico/validation/networkconfig.go

```diff
@@ -166,22 +183,24 @@ func ValidateIPAutodetectionMethod(method string, fldPath *field.Path) field.Err
 }
 
 // IsValidMTU checks if the provided MTU is a valid positive integer.
-func IsValidMTU(mtu string) error {
+func IsValidMTU(mtu string, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
 	if mtu == "" {
-		return fmt.Errorf("MTU cannot be empty")
+		allErrs = append(allErrs, field.Invalid(fieldPath, mtu, "MTU cannot be empty"))
+		return allErrs
 	}
 	mtuInt, err := strconv.Atoi(mtu)
 	if err != nil {
-		return fmt.Errorf("invalid MTU: %w", err)
+		allErrs = append(allErrs, field.Invalid(fieldPath, mtu, fmt.Sprintf("invalid MTU: %v", err)))
 	}
-
-	if mtuInt <= 0 {
-		return fmt.Errorf("MTU must be a positive integer")
+	if mtuInt < 0 {
+		allErrs = append(allErrs, field.Invalid(fieldPath, mtu, "MTU must be a positive integer"))
 	}
```

Comment:
Would we not still return in case of errors? A nice implementation of `Atoi(...)` might not return a value if there is an error, but a not-so-nice implementation might so you could end up with a completely unrelated error because `Atoi(...)` returned a negative number and an error.

```suggestion
		return append(allErrs, field.Invalid(fieldPath, mtu, fmt.Sprintf("invalid MTU: %v", err)))
	}

	if mtuInt < 0 {
		return append(allErrs, field.Invalid(fieldPath, mtu, "MTU must be a positive integer"))
	}
```